cpp/src/arrow/util/tracing_internal.h

```diff
@@ -114,23 +117,38 @@ class SpanImpl {
 opentelemetry::trace::StartSpanOptions SpanOptionsWithParent(
     const util::tracing::Span& parent_span);
 
-#define START_SPAN(target_span, ...)                                                \
-  auto opentelemetry_scope##__LINE__ =                                              \
-      ::arrow::internal::tracing::GetTracer()->WithActiveSpan(                      \
-          target_span                                                               \
-              .Set(::arrow::util::tracing::Span::Impl{                              \
-                  ::arrow::internal::tracing::GetTracer()->StartSpan(__VA_ARGS__)}) \
-              .span)
+#define FIRST_ARG(first, ...) first
+
+#define START_COMPUTE_SPAN(target_span, ...) \
+  START_SPAN(target_span, __VA_ARGS__);      \
+  START_SPAN(target_span, FIRST_ARG(__VA_ARGS__), {GET_MEMORY_POOL_INFO});
```

Comment:
I don't really understand why this starting two spans (or starting the same span twice)? @lidavidm Does this look right?

Reply:
Yeah, this seems to create two spans, and immediately closes the first one. I don't think this is what we want.

Reply:
Ah I see now. START_SPAN is just activating the span. I think what you want is a macro to attach a new attribute to the span, not repeating the START_SPAN macro.

Reply:
I tried, to attach the new attribute to the current __VA_ARGS_  but there's no solution on how to remove the curly brackets `{}` and insert the new attribute and then pass it to the START_SPAN, I tried even create a std::vector<> insert the new attribute and convert it to std::initializer_list or create the OT::nostd::span<> manually but nothing worked

Reply:
Just call `span->SetAttribute`. Look at the header file or docs. https://github.com/open-telemetry/opentelemetry-cpp/blob/main/api/include/opentelemetry/trace/span.h#L46-L50